front50-web/src/main/groovy/com/netflix/spinnaker/front50/controllers/v2/ProjectsController.java

```diff
@@ -0,0 +1,244 @@
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.netflix.spinnaker.front50.controllers.v2;
+
+import static java.lang.String.format;
+
+import com.google.common.base.Splitter;
+import com.google.common.base.Strings;
+import com.netflix.spectator.api.Registry;
+import com.netflix.spinnaker.front50.UntypedUtils;
+import com.netflix.spinnaker.front50.exception.NotFoundException;
+import com.netflix.spinnaker.front50.exceptions.InvalidRequestException;
+import com.netflix.spinnaker.front50.model.SearchUtils;
+import com.netflix.spinnaker.front50.model.project.Project;
+import com.netflix.spinnaker.front50.model.project.ProjectDAO;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import java.util.*;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import javax.servlet.http.HttpServletResponse;
+import lombok.Value;
+import org.springframework.context.MessageSource;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequestMapping(value = "/v2/projects", produces = MediaType.APPLICATION_JSON_VALUE)
+@Api(value = "projects", description = "Project API")
+public class ProjectsController {
+
+  private static final Splitter COMMA_SPLITTER = Splitter.on(',');
+
+  private MessageSource messageSource;
+  private ProjectDAO projectDAO;
+  private Registry registry;
+
+  @RequestMapping(value = "/search", method = RequestMethod.GET)
+  @ApiOperation(
+      value = "",
+      notes =
+          "Search for projects given one or more attributes.\n\n- /search?q=ProjectName\n- /search?q=ApplicationName\n")
+  public Set<Project> search(@RequestParam("q") final String query) {
+    return projectDAO.all().stream()
+        .filter(
+            p ->
+                p.getName().toLowerCase().contains(query.toLowerCase())
+                    || p.getConfig().getApplications().stream()
+                        .anyMatch(it -> it.toLowerCase().contains(query.toLowerCase())))
+        .collect(Collectors.toSet());
+  }
+
+  @ApiOperation(
+      value = "",
+      notes =
+          "Fetch all projects.\n\n    Support filtering by one or more attributes:\n    - ?name=projectName\n    - ?email=my@email.com")
+  @RequestMapping(method = RequestMethod.GET)
+  public Set<Project> projects(
+      @RequestParam(value = "pageSize", required = false) Integer pageSize,
+      @RequestParam Map<String, String> params) {
+    params.remove("pageSize");
+    Set<Project> projects =
```

Comment:
IntelliJ didn't complain about it so I figured it would be OK, but I've wrapped `projectDAO.all()` into `new HashSet<>(projectDAO.all())` to be safe. Since `all()` is defined by `ItemDAO<T>`, I'm not comfortable in changing the signature of the method without affecting other places.